OmgvaPOS/src/BusinessManagement/Services/BusinessService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using OmgvaPOS.BusinessManagement.DTOs;
+using OmgvaPOS.BusinessManagement.Mappers;
+using OmgvaPOS.BusinessManagement.Repository;
+
+namespace OmgvaPOS.BusinessManagement.Services
+{
+    public class BusinessService(IBusinessRepository businessRepository) : IBusinessService
+    {
+        private readonly IBusinessRepository _businessRepository = businessRepository;
+        public List<BusinessDTO> GetBusinesses() {
+            return _businessRepository.GetBusinesses().Select(b => b.ToBusinessDTO()).ToList();
+        }
+        public BusinessDTO GetBusiness(long id) {
+            return _businessRepository.GetBusiness(id).ToBusinessDTO();
+        }
+        public BusinessDTO CreateBusiness(CreateBusinessRequest request) {
+            return _businessRepository.CreateBusiness(request.ToBusiness()).ToBusinessDTO();
+        }
+        public bool UpdateBusiness(BusinessDTO business) {
+            return _businessRepository.UpdateBusiness(business);
```

Comment:
Please map businessDTO -> business model before passing it to repository, so that repository would have minimal logic inside

Reply:
The problem is that for this conversion I need to fetch it from the repository first to map the DTO on top of the business as it can have null fields.

Reply:
Do you think it would be better to fetch in the service?

Reply:
Yes, in my opinion we should add this fetch / update logic to service, repository should not need to know anything about DTOs

Reply:
Sure, will redo

Reply:
Done